### ISSUE TITLE
macOS: Debug fails when running though PyCharm

### ISSUE THREAD

**Issue by hhusum** [tensorflow/tensorflow#18490]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: yes
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: macOS High Sierra 10.13.4
- **TensorFlow installed from (source or binary)**: binary, pip
- **TensorFlow version (use command below)**: 1.7
- **Python version**: 3.6.4 Anaconda, Inc.
- **Bazel version (if compiling from source)**: n/a
- **GCC/Compiler version (if compiling from source)**: n/a
- **CUDA/cuDNN version**: n/a
- **GPU model and memory**: Intel Iris Pro 1536 MB
- **Exact command to reproduce**: n/a

### Describe the problem
Trying to debug a tf.Estimator script, using PyCharm as IDE. For this I use the tfdbg.LocalCLIDebugHook. When running the script through terminal debugging works fine, but through PyCharm i get a curses error:

`_curses.error: setupterm: could not find terminal`

I know this is more a PyCharm issue than a tf issue, but if there is anything simple you could do to mitigate this, it would be nice.

### Source code / logs
File "/Users/harald/anaconda3/envs/thesis/lib/python3.6/site-packages/tensorflow/python/estimator/estimator.py", line 355, in train
    loss = self._train_model(input_fn, hooks, saving_listeners)
  File "/Users/harald/anaconda3/envs/thesis/lib/python3.6/site-packages/tensorflow/python/estimator/estimator.py", line 903, in _train_model
    _, loss = mon_sess.run([estimator_spec.train_op, estimator_spec.loss])
  File "/Users/harald/anaconda3/envs/thesis/lib/python3.6/site-packages/tensorflow/python/training/monitored_session.py", line 546, in run
    run_metadata=run_metadata)
  File "/Users/harald/anaconda3/envs/thesis/lib/python3.6/site-packages/tensorflow/python/training/monitored_session.py", line 1022, in run
    run_metadata=run_metadata)
  File "/Users/harald/anaconda3/envs/thesis/lib/python3.6/site-packages/tensorflow/python/training/monitored_session.py", line 1113, in run
    raise six.reraise(*original_exc_info)
  File "/Users/harald/anaconda3/envs/thesis/lib/python3.6/site-packages/six.py", line 693, in reraise
    raise value
  File "/Users/harald/anaconda3/envs/thesis/lib/python3.6/site-packages/tensorflow/python/training/monitored_session.py", line 1098, in run
    return self._sess.run(*args, **kwargs)
  File "/Users/harald/anaconda3/envs/thesis/lib/python3.6/site-packages/tensorflow/python/training/monitored_session.py", line 1162, in run
    feed_dict, options)
  File "/Users/harald/anaconda3/envs/thesis/lib/python3.6/site-packages/tensorflow/python/training/monitored_session.py", line 1188, in _call_hook_before_run
    request = hook.before_run(run_context)
  File "/Users/harald/anaconda3/envs/thesis/lib/python3.6/site-packages/tensorflow/python/debug/wrappers/hooks.py", line 106, in before_run
    on_run_start_request)
  File "/Users/harald/anaconda3/envs/thesis/lib/python3.6/site-packages/tensorflow/python/debug/wrappers/local_cli_wrapper.py", line 255, in on_run_start
    self._prep_cli_for_run_start()
  File "/Users/harald/anaconda3/envs/thesis/lib/python3.6/site-packages/tensorflow/python/debug/wrappers/local_cli_wrapper.py", line 277, in _prep_cli_for_run_start
    self._run_cli = ui_factory.get_ui(self._ui_type)
  File "/Users/harald/anaconda3/envs/thesis/lib/python3.6/site-packages/tensorflow/python/debug/cli/ui_factory.py", line 61, in get_ui
    return curses_ui.CursesUI(on_ui_exit=on_ui_exit, config=config)
  File "/Users/harald/anaconda3/envs/thesis/lib/python3.6/site-packages/tensorflow/python/debug/cli/curses_ui.py", line 287, in __init__
    self._screen_init()
  File "/Users/harald/anaconda3/envs/thesis/lib/python3.6/site-packages/tensorflow/python/debug/cli/curses_ui.py", line 400, in _screen_init
    self._stdscr = curses.initscr()
  File "/Users/harald/anaconda3/envs/thesis/lib/python3.6/curses/__init__.py", line 30, in initscr
    fd=_sys.__stdout__.fileno())
_curses.error: setupterm: could not find terminal

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Bazel version
CUDA/cuDNN version
Exact command to reproduce

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by hhusum** [comment]
Yes, it is.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @aselle: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @aselle: It has been 29 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @aselle: It has been 44 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @aselle: It has been 59 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @aselle: It has been 74 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by aselle** [comment]
tfdbg requires curses it looks like. you may need to use the command line to do debugging. Assigning to the author of tfdbg, for additional info. @caisq could you provide more support.

**Comment by caisq** [comment]
@hhusum @aselle Yes, this error occurs because tfdbg uses `curses` to manage the terminal UI by default. PyCharm is an environment that does not have a tty for curses to use. To work around this, set the `ui_type` kwarg to `readline` when using `LocalCLIDebugWrapperSession` or `LocalCLIDebugHook`.
